MetaMask/metamask-extension
# Integrate LavaDome private key protection

## **Description**



New [LavaDome](https://github.com/LavaMoat/LavaDome) tool can help to better hide the private key from potential malicious code running in the app when the user asks to see it.

## **Related issues**

None

## **Manual testing steps**

This PR changes the "show private key" dialog, not in terms of appearance, and not in terms of interaction but only in terms of "behind the scenes" security, so nothing should change in terms of manual interaction with this feature - that is the most important thing to check here QA-wise.

One thing that is changed, is that the private key text is no longer selectable as it was before, and now the only way to copy it is to use the copy-to-clipboard button.

That is on purpose, allowing selection of sensitive information such as the private key is unsafe, and disablement of selection capabilities of sensitive info is done intentionally to enhance security.

## **Screenshots/Recordings**



### **Before**

![Screenshot 2024-01-10 at 22 43 55](https://github.com/MetaMask/metamask-extension/assets/13243797/93eec4f4-9be7-4440-930c-55bce8c4d05e)

### **After**

<img src="https://github.com/MetaMask/metamask-extension/assets/13243797/bc8ec626-952a-42b4-893a-5fa15b38be4a" alt="key in image isn't real" />

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
